src/sentry/api/endpoints/organization_transaction_anomaly_detection.py

```diff
@@ -0,0 +1,92 @@
+from datetime import datetime
+
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.bases import OrganizationEventsEndpointBase
+from sentry.api.utils import get_date_range_from_params
+from sentry.search.events.builder import QueryBuilder
+from sentry.snuba.dataset import Dataset
+
+
+class OrganizationTransactionAnomalyDetectionEndpoint(OrganizationEventsEndpointBase):
+    def get(self, request: Request, organization) -> Response:
+        start, end = get_date_range_from_params(request.GET)
+        query_start, query_end, granularity = self.map_snuba_queries(
+            start.timestamp(), end.timestamp()
+        )
+
+        # remove unused and overwrite relevant time params
+        for param in ["statsPeriod", "statsPeriodStart", "statsPeriodEnd"]:
+            request.GET.pop(param, None)
+        request.GET[start] = query_start
+        request.GET[end] = query_end
+
+        params = self.get_snuba_params(request, organization)
```

Comment:
I'm a little confused why the start&end need to be modified before calling `get_snuba_params`

Reply:
It's because the dataset needed for visualization will not be the same as the dataset needed for anomaly detection (e.g. frontend renders anomaly detection from 1/1-1/7, but we need to pull data from 12/1-1/7 to run algorithm)

Reply:
Ah in that case the cleaner thing to do here miight be to get_snuba_params first, then adjust the start in the resulting params accordingly.